apps/cow-tools/src/app/milkman/(components)/orderForm/Resume.tsx

```diff
@@ -87,38 +87,59 @@ export function OrderResume({
       <div className="flex flex-col">
         {fieldsToDisplay.map((field) => {
           const value = data[field.key];
-          if (field.key === "tokenSell") {
-            return (
-              value && (
-                <span key={field.key}>
-                  {field.label}: {value.symbol} (
-                  {truncateAddress(data.tokenSell.address)})
-                </span>
-              )
-            );
-          }
-          if (field.key === "tokenBuy") {
-            return (
-              value && (
-                <span key={field.key}>
-                  {field.label}: {value.symbol} (
-                  {truncateAddress(data.tokenBuy.address)})
-                </span>
-              )
-            );
+          switch (field.key) {
```

Comment:
these switch case statements within components render make it so hard to reason about the whole component logic -- I trust your code here works but `would` make an effort to modularize a bit more this code and avoid inline logic, perhaps you could start taking a read at code reviews in other projects as well to see how other folks have handled this